docs/releases/3.0.txt

```diff
@@ -416,6 +416,9 @@ Miscellaneous
   when ``doseq=False``, rather than iterating them, bringing it into line with
   the standard library :func:`urllib.parse.urlencode` function.
 
+* ``intword`` template filter now translates ``1.0`` as a singular phrase and
+  all other numeric values as plural. This may be incorrect for some languages.
```

Comment:
Sorry I'm late to the party with this one, but...

What is to stop us defining a new constant in `django/django/conf/locale/*/formats.py` to select between `round_away_from_one()` and an additional `round_toward_one()` such that this can be addressed on a per-locale basis?

(Or a combined `round_singular_plural()`, or something named better, that uses the active locale to determine what to do...)

Reply:
I'm open to this. But I don't know enough about non-English languages to know how this should look. Can you flesh out your idea a bit on the different cases we would need to handle? In other words, what might this constant be called, what values could it have, and how would those values affect the rounding of `round_singular_plural`?

What is the difference between `LANG_INFO` and `formats.py` for this purpose? Is one better suited?

Reply:
I'm not sure if we will be able to handle this properly for all languages, e.g. for Polish :poland: , both the previous and current form are bad, i.e. all fractional numbers should have a separate form `miliona` (see [comments](https://bugs.python.org/issue28692)). To be honest I'm fine with that.

I don't think that we should create :building_construction:  (and maintain) a separate mechanism just for this single template filter (\cc @claudep).

Reply:
> Can you flesh out your idea a bit on the different cases we would need to handle?

Ooh. I wouldn't profess to be an expert in any way :no_mouth: I'm just aware that this PR solves part of the problem, but states that it's going to to be wrong for a bunch of people. Also we are rounding away from one and @felixmm mentioned that Polish will now output the wrong string - hence the "rounding toward one" I suggested, although that may have been presumptuous...

> In other words, what might this constant be called, what values could it have, and how would those values affect the rounding of `round_singular_plural`?

I was initially thinking of something that was just a boolean - whether it rounded away from or toward one, but I can see that this may potentially be more complicated. I didn't put that much thought into it. These links, however, could be a useful starting point:
- https://www.unicode.org/cldr/charts/33/supplemental/language_plural_rules.html
- https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html
- https://en.wikipedia.org/wiki/Plural#Usage_of_the_plural

> What is the difference between `LANG_INFO` and `formats.py` for this purpose? Is one better suited?

`LANG_INFO` only seems to contain information on language name, code and direction, whereas `formats.py` contains things such as date, time, and number formats. This sort of feels like it is a formatting option...

Reply:
> I don't think that we should create  (and maintain) a separate mechanism just for this single template filter

That's fair. Sorry if I'm being awkward poking my nose in!

Reply:
@pope1ni Don't worry, that's just my opinion. We can decide otherwise :man_shrugging: .

Reply:
> We are not going to fix gettext own limitations here.

After reading a bit more into this, I agree.